src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatResponseExtensions.cs

```diff
@@ -219,6 +339,8 @@ internal static void CoalesceContent(IList<AIContent> contents)
                 return content;
             });
 
+        CoalesceImageDataContent(contents);
```

Comment:
I have replacement semantics for images, which may be non-contiguous.  For example, when streaming images, you have 2 iterations of 2 images, they will be interleaved.  The subsequent images are more detailed versions of prior ones with the same name.

I might be able to get this to work using Coalesce<DataContent> if I record the names I see along the way in the canMerge callback (and just ignore previous parameter).  Let me know if you would prefer that.